test/end-to-end/specs/blueprints.test.js

```diff
@@ -0,0 +1,103 @@
+const faker = require('faker');
+const addContext = require('mochawesome/addContext');
+
+const blueprintsPage = require('../pages/blueprints.page');
+const Blueprint = require('../components/Blueprint.component');
+
+describe('Blueprints Page', function(){
+  before(function(){
+    browser.login();
+  });
+
+  describe('Visual Regression', function(){
+    before(function() {
+      if(['chrome', 'MicrosoftEdge'].includes(browser.desiredCapabilities.browserName)) {
+        addContext(this, `skip visual regression on ${browser.desiredCapabilities.browserName} \
+          because element screenshoot is not well supported by ${browser.desiredCapabilities.browserName}`);
+        this.skip();
+      }
+    })
+    it('Toolbar should look good', function(){
+      addContext(this, 'do screenshoot on toobar');
+      const report = browser.checkElement('.toolbar-pf');
+      browser.expectGoodThings(report);
+    });
+
+    it('Blueprints list view should look good', function(){
+      addContext(this, 'do screenshoot on blueprint list view');
+      const report = browser.checkElement('.list-view-pf-view');
+      browser.expectGoodThings(report);
+    });
+  });
+
+  describe('New Blueprint', function(){
+    let name, description, blueprintComponent;
+    beforeEach(function(){
+      // blueprint name cannot contain space due to issue https://github.com/weldr/welder-web/issues/317
+      // use lorem.slug() ("a-accusantium-repudiandae") instead of lorem.words() ("nulla placeat qui")
+      name = faker.lorem.slug();
+      description = faker.lorem.sentence();
```

Comment:
TBH I'm not a big fan of such fancy stuff; just static strings are easier to control, easier to understand as a reader, and more robust for repeated debugging.

Reply:
@martinpitt, the blueprint name has to be random if we run test case in parallel. Each test will run on different blueprint.